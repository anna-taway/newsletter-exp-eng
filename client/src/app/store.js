import { configureStore } from "@reduxjs/toolkit";
import registrationReducer from "../features/userRegistration/userRegistrationSlice";
import productsReducer from '../features/products/productsSlice';


const store = configureStore({
	reducer: {
		registration: registrationReducer,
		products: productsReducer
	},
});

export default store;