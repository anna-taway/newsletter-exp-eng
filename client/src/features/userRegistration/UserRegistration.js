import React, { useState } from "react";
import { useSelector } from 'react-redux';
import { makeStyles } from '@material-ui/core/styles';
import { useDispatch } from "react-redux";
import { newUserRegistration } from './userRegistrationAction';

import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Link from '@material-ui/core/Link';
import Box from '@material-ui/core/Box';
import Alert from '@material-ui/lab/Alert';

const initialState = {
  username: '',
};

const useStyles = makeStyles((theme) => ({
  form: {
    width: '100%',
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  box: {
    width: '100%'
  }
}));

const UserRegistration = () => {
  
  const classes = useStyles();
  const dispatch = useDispatch();
  const [newUser, setNewUser] = useState(initialState);

  const isLoading = useSelector(({registration}) => registration.isLoading);
  const registrationStatus = useSelector(({registration}) => registration.status);

  const handleOnChange = (e) => {
    const { name, value } = e.target;

    setNewUser({ ...newUser, [name]: value });

  };

  const handleOnSubmit = (e) => {
    e.preventDefault();
    const { username, password } = newUser;

    const newRegistration = {
      username: username,
      password: password
    };
    dispatch(newUserRegistration(newRegistration));
  };

  return (
    <React.Fragment>
      <form className={classes.form} noValidate onSubmit={handleOnSubmit}>
        <TextField
          variant="outlined"
          margin="normal"
          required
          fullWidth
          id="username"
          label="Username"
          name="username"
          autoComplete="username"
          autoFocus
          value={newUser.username}
          onChange={handleOnChange}
        />
        <FormControlLabel
          control={<Checkbox value="remember" color="primary" />}
          label="Remember me"
        />
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
          >
            Sign In
          </Button>
          <Grid container>
            <Grid item xs>
              <Link href="#" variant="body2">
                Forgot password?
              </Link>
            </Grid>
            <Grid item>
              <Link href="#" variant="body2">
                {"Don't have an account? Sign Up"}
              </Link>
            </Grid>
          </Grid>
        </form>
      <Box className={classes.box} mt={6}>
        {isLoading ? <></> : (
          registrationStatus === 'success' ? 
            <Alert severity="success">Successful registration</Alert> : 
          <Alert severity="error">Please fill the field</Alert>) 
        }
      </Box>
    </React.Fragment>
  );
}

export default UserRegistration;




