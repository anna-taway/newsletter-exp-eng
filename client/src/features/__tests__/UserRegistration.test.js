import React from "react";
import UserRegistration from "../userRegistration/UserRegistration";
import { create } from "react-test-renderer";
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'

import { Provider } from 'react-redux';
import store from '../../app/store';

describe("UserRegistration component", () => {
  test("UserRegistration component", () => {
    const history = createMemoryHistory()
    const userRegistration = create(
        <Provider store={store}>
            <Router history={history}>
                <UserRegistration />
            </Router>
        </Provider>
    )
    expect(userRegistration.toJSON()).toMatchSnapshot();
  });
});