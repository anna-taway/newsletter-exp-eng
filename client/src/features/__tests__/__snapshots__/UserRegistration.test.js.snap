// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRegistration component UserRegistration component 1`] = `
Array [
  <form
    className="makeStyles-form-1"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
        data-shrink={false}
        htmlFor="username"
        id="username-label"
      >
        Username
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="username"
          autoFocus={true}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="username"
          name="username"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-6"
          >
            <span>
              Username
               *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <label
      className="MuiFormControlLabel-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-8 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            className="PrivateSwitchBase-input-11"
            data-indeterminate={false}
            onChange={[Function]}
            type="checkbox"
            value="remember"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Remember me
      </span>
    </label>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-2 MuiButton-containedPrimary MuiButton-fullWidth"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Sign In
      </span>
    </button>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
          href="#"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Forgot password?
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
          href="#"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Don't have an account? Sign Up
        </a>
      </div>
    </div>
  </form>,
  <div
    className="MuiBox-root MuiBox-root-12 makeStyles-box-3"
  >
    <div
      className="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
      role="alert"
    >
      <div
        className="MuiAlert-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        className="MuiAlert-message"
      >
        Please fill the field
      </div>
    </div>
  </div>,
]
`;
